### PR TITLE
feat(query_log): Hackathon access metrics from query_log in PostHog

### PR DESCRIPTION
**Pull request by anirudhpillai**
## Problem
Hackathon - Improve monitoring
Don't have query metrics in PostHog

## Changes
Expose query_log as a lazy table so we can use it to create insights with monitoring metrics.

image

## Does this work well for both Cloud and self-hosted?
N/A

### CODE REVIEW

File: posthog/hogql/database/database.py
Comment (Gilbert09) [inline posthog/hogql/database/database.py:118]: We should only be adding this class attribute when we've checked that we're allowed to. Likewise with the raw one too - we don't always check `posthog_tables` before trying to access the table

File: posthog/hogql/database/database.py
Comment (Gilbert09) [inline posthog/hogql/database/database.py:497]: We also wanna check what region we're in here too - e.g. team 2 in EU isn't us

File: posthog/hogql/printer.py
Comment (Gilbert09) [inline posthog/hogql/printer.py:57]: Same with checking region here (or do what Ben said about the user roles)

File: posthog/constants.py
Comment (webjunkie) [inline posthog/constants.py:322]: Should these be constants or rather settings (thinking of self hosting folks)?

Comment (Gilbert09) [reply posthog/constants.py:322]: Or better still, injected via env vars from charts

File: posthog/hogql/database/database.py
Comment (Gilbert09) [inline posthog/hogql/database/database.py:246]: Not sure if `InternalDatabase` is the correct naming here (nor whether subclassing `Database`. I was imagining you using `setattr(..)` to add the new tables to an existing `Database` object

**Review by Gilbert09 — APPROVED** [review]
Im fine with this, but best to get @mariusandra to double check some of the team id guarding stuff before merging please

File: posthog/hogql/database/database.py
Comment (anirudhpillai) [reply posthog/hogql/database/database.py:246]: Issue with `setattr` was that these are class variables so if we set them once, then they'll persist on the class. Separate classes solves this issue

**Comment by anirudhpillai** [comment]
Going to hold off on this and try to get query_log in as a data warehouse table as when we start persisting it, it will flow through S3. So doesn't make sense to override the team id logic/add more complexity, if it'll just be a temporary solution :(

**Pull request closed without merging**
